process/sync/baseForkDetector.go

```diff
@@ -70,6 +69,9 @@ func (bfd *baseForkDetector) checkBlockBasicValidity(header data.HeaderHandler,
 	if roundDif < nonceDif {
 		return ErrHigherNonceInBlock
 	}
+	if bfd.blackListHandler.Has(string(header.GetPrevHash())) {
```

Comment:
I think it would be good to add to the blackListHandler.

Reply:
I have tried this, but in some system tests there were some situations, where I have suspected that because I have added here the received headers which are bound with a previous black listed header, something went wrong with the synchronization. We need more tests to be sure about this action. I would put here a TODO